packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -134,51 +134,78 @@ export const addItemToCart = (payload = {}) => {
     };
 };
 
+/**
+ * Applies changes in options/quantity to a cart item.
+ *
+ * @param payload.cartItemId {Number} the id of the cart item we are updating
+ * @param payload.item {Object} the new configuration item if changes are selected.
+ * @param payload.quantity {Number} the quantity of the item being updated
+ * @param payload.productType {String} 'ConfigurableProduct' or other.
+ */
 export const updateItemInCart = (payload = {}) => {
-    const { cartItemId, fetchCartId, item } = payload;
+    const {
+        cartItemId,
+        fetchCartId,
+        item,
+        productType,
+        quantity,
+        updateItem
+    } = payload;
     const writingImageToCache = writeImageToCache(item);
 
     return async function thunk(dispatch, getState) {
         await writingImageToCache;
         dispatch(actions.updateItem.request(payload));
 
-        const { cart, user } = getState();
+        const {
+            cart: { cartId },
+            user
+        } = getState();
         const { isSignedIn } = user;
 
         try {
-            let cartEndpoint;
-
-            if (!isSignedIn) {
-                const { cartId } = cart;
-
-                if (!cartId) {
-                    const missingCartIdError = new Error(
-                        'Missing required information: cartId'
-                    );
-                    missingCartIdError.noCartId = true;
-                    throw missingCartIdError;
-                }
-
-                cartEndpoint = `/rest/V1/guest-carts/${cartId}/items/${cartItemId}`;
+            if (productType === 'ConfigurableProduct') {
+                await dispatch(
```

Comment:
We _must_ remove and then add. If changes are made to quantity only and we add first then remove we end up deleting the entire cart item.